examples/custom-controller/deployment.yaml

```
@@ -0,0 +1,65 @@
+---
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: custom-controller-sa
+  namespace: agones-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: custom-controller-role
+rules:
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["create", "patch"]
+  - apiGroups: ["agones.dev"]
+    resources: ["gameservers"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: custom-controller-access
+subjects:
+  - kind: ServiceAccount
+    name: custom-controller-sa
+    namespace: agones-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: custom-controller-role
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: custom-controller
+  namespace: agones-system
+spec:
+  replicas: 2
```

To you question in another box, this should be 1

```suggestion
  replicas: 1
```

Good question - unless a controller has leader election you always want one of them, otherwise they can run into each other, trying to do the same things (this might be why you end up with so many conflicts?)

---

Adding leader election to the example would be a good follow up task. It's worth filing a separate issue for.

---

Sound good. Filed this issue - https://github.com/googleforgames/agones/issues/3694

---

Would you mind taking a look at this log: https://gist.github.com/Kalaiselvi84/db200f984b4102ff2de7f7baf93b13a7? I'll proceed with pushing the image to production once you confirm.